### PR TITLE
Implementing manure treatment Composting method

### PR DESCRIPTION
**Pull request by allisterakun**
This PR implements the _Composting_ manure treatment method:
- Created a `Composting` class:
    ```
    Class for managing and simulating the composting process of manure treatment.

    This class simulates the composting process by considering various factors like weather,
    manure characteristics, and composting configurations. It provides methods for daily
    update of compost characteristics such as methane emissions, nitrogen content, and
    carbon decomposition. The calculations are based on standard composting models and
    environmental factors.

    Attributes
    ----------
    weather : Weather
        The current weather conditions.
    time : Time
        The current time information.
    manure_treatment_config : ManureTreatmentConfig
        Configuration settings for manure treatment.
    composting_type : CompostingType
        The type of composting process being used.

    Methods
    -------
    _daily_update_helper() -> ManureTreatmentDailyOutput:
        Performs daily updates to the compost characteristics.

    calc_methane_emission(*args, **kwargs) -> float:
        Calculates the methane emissions for the current day.

    calc_ammonia_emission(*args, **kwargs) -> float:
        Calculates the total nitrogen loss to ammonia emission for the current year.

    _calculate_methane_conversion_factor() -> float:
        Returns the methane conversion factor based on composting type and temperature.

    _calculate_max_microbial_decomposition_rate() -> float:
        Calculates the maximum microbial decomposition rate for the current day.

    _calculate_slow_microbial_decomposition_rate() -> float:
        Calculates the slow microbial decomposition rate for the current day.

    _calculate_carbon_decomposition_rate() -> float:
        Calculates the carbon decomposition rate for the current day.

    _calculate_anaerobic_coefficient() -> float:
        Calculates the anaerobic coefficient for the composting process.

    _calculate_carbon_decomposition(total_solid: float) -> float:
        Computes the total carbon decomposition for the current day.

    _calculate_dry_matter_loss(methane_emission: float, carbon_decomposition: float) -> float:
        Calculates the total dry matter loss for the current day.

    _calculate_Nitrogen_loss_to_leaching() -> float:
        Computes the nitrogen loss due to leaching for the current year.

    _calculate_Nitrogen_loss_to_direct_Nitrous_Oxide_Emission() -> float:
        Calculates the nitrogen loss through direct nitrous oxide emission for the current year.

    _calculate_Nitrogen_loss_to_indirect_Nitrous_Oxide_Emission() -> float:
        Calculates the nitrogen loss through indirect nitrous oxide emission for the current year.

    _calculate_total_Nitrogen_mass() -> float:
        Computes the total mass of nitrogen in the manure-bedding mix for the current year.

    _calculate_organic_Nitrogen_mass() -> float:
        Calculates the mass of organic nitrogen in the manure-bedding mix for the current year.

    _calculate_inorganic_Nitrogen_mass() -> float:
        Calculates the mass of inorganic nitrogen in the manure-bedding mix for the current year.

    _calculate_ammonium_mass() -> float:
        Computes the mass of ammonium in the manure-bedding mix for the current year.

    ```
- Created a DefaultEnum class `CompostingTypes`
    ```
    This is an Enum class that represents different types of composting.

    Attribute
    ----------
    INTENSIVE_WINDROW : str
        Intensive windrow.
    PASSIVE_WINDROW : str
        Passive windrow.
    STATIC_PILE: str
        Static pile
    ```
- Added composting scenarios in `manure_management.json`
- Added composting-related config variables in metadata:
    ```
           "composting_type": {
              "type": "string",
              "description": "Composting Type -- The type of composting.",
              "pattern": "^(intensive windrow|passive windrow|static pile)$",
              "default": "intensive windrow"
            },
            "last_compost_turning_or_addition": {
              "type": "number",
              "description": "The number of days since the last turning or addition event (day)",
              "default": 1
            },
    ```

### CODE REVIEW

File: RUFAS/routines/manure/manure_treatments/composting.py
Comment (KFosterReed) [inline RUFAS/routines/manure/manure_treatments/composting.py]: Would it make sense to move temperature constants like this to the GasEmissionsConstants.py file? I think it may be clearer and easier to modify in the long run

File: RUFAS/routines/manure/manure_treatments/composting.py
Comment (KFosterReed) [inline RUFAS/routines/manure/manure_treatments/composting.py]: This is a correct reflection of the equation in the composting document but just confirmed with April and Kaitlyn that we should not be subtracting the indirect N2O emissions from this equation. Please delete the Nitrogen_loss_to_indirect_N2O_emission from this equation

File: RUFAS/routines/manure/manure_treatments/composting.py
Comment (KFosterReed) [inline RUFAS/routines/manure/manure_treatments/composting.py]: Can we please remove all methods for indirect nitrous oxide emission? After reviewing with the manure SME's, this is actually something that happens off farm and should be estimated post RuFaS biophysical simulation. 

Sorry to make you undo your great work!

Also, as a side note, we will update the composting design doc accordingly.

File: RUFAS/routines/manure/manure_treatments/manure_treatment_configs.py
Comment (KFosterReed) [inline RUFAS/routines/manure/manure_treatments/manure_treatment_configs.py:96]: We need to add the descriptions of all new attributes to the docstrings but before we do that, I think many of these need to be move elsewhere. 

I am cross checking with the design doc but for now, I am confident that the following should be moved to gas_emission_constants file: 

```
 maximum_methane_producing_capacity: float = 0.24
    effectiveness_of_microbial_decomposition_rate: float = 2.37e-3
    decomposition_temperature: float = 60
    first_order_decaying_coefficient: float = 0.1
   
 mole_fraction_of_oxygen: float = 0.15
    half_saturation_constant: float = 0.02
    ambient_air_mole_fraction_of_oxygen: float = 0.21
 effect_of_moisture_on_microbial_decomposition: float = 0.65
```
Also check the half saturation constant, I think it is off by a digit (0.002 instead of 0.02)

Comment (KFosterReed) [reply RUFAS/routines/manure/manure_treatments/manure_treatment_configs.py:96]: @allisterakun sorry for the confusion, I had meant to remove ` last_compost_turning_or_addition: int = 1` from the above list before posting. Also confirming that the above list should be moved to the constants section as well as the following: 

`  lag_time: int = 2`

`proportion_of_carbon_available_in_manure: float = 0.5`

`proportion_of_carbon_available_in_bedding: float = 0.35`

Thank you!

File: RUFAS/time.py
Comment (ltp23) [inline RUFAS/time.py]: Could this be even more exact by using `==` instead of `>=`?

Comment (allisterakun) [reply RUFAS/time.py]: This function is no longer needed as we are not doing any annual updates. Will delete this function in my next commit!

File: RUFAS/routines/manure/manure_treatments/manure_treatment_annual_output.py
Comment (ltp23) [inline RUFAS/routines/manure/manure_treatments/manure_treatment_annual_output.py]: This class doesn't have all the attributes specified by the LiquidManurePortionProtocol. If it is working okay, then maybe you don't need it to.

Comment (allisterakun) [reply RUFAS/routines/manure/manure_treatments/manure_treatment_annual_output.py]: This class is also deleted since we are not doing annual updates anymore!

File: RUFAS/routines/manure/manure_treatments/manure_treatment_annual_output.py
Comment (ltp23) [inline RUFAS/routines/manure/manure_treatments/manure_treatment_annual_output.py]: If you can remove this line here and in some other files involved in this PR, that would be great. Because we are assuming Python 3.10+ is used, this backward-compatible guard is no longer needed.

Comment (allisterakun) [reply RUFAS/routines/manure/manure_treatments/manure_treatment_annual_output.py]: Ok! Will do that!

Comment (allisterakun) [reply RUFAS/routines/manure/manure_treatments/manure_treatment_annual_output.py]: Done!

Comment (allisterakun) [reply RUFAS/routines/manure/manure_treatments/manure_treatment_annual_output.py]: I had to keep this line in `RUFAS/routines/manure/manure_treatments/manure_treatment_annual_output.py`, as it is causing issues. All the other appearances are deleted!

File: RUFAS/routines/manure/manure_treatments/composting.py
Comment (ltp23) [inline RUFAS/routines/manure/manure_treatments/composting.py]: I think these constants should be moved into one of the constants files.

File: RUFAS/routines/manure/manure_treatments/composting.py
Comment (ltp23) [inline RUFAS/routines/manure/manure_treatments/composting.py]: I think to keep the names more consistent, it's probably better to use all lower-case.

File: RUFAS/routines/manure/manure_treatments/composting.py
Comment (ltp23) [inline RUFAS/routines/manure/manure_treatments/composting.py]: Please use a better name for `o2` or `O2` here because it could be confused with `02` if someone can't tell the difference between the letter and the digit.

File: RUFAS/routines/manure/manure_treatments/composting.py
Comment (allisterakun) [reply RUFAS/routines/manure/manure_treatments/composting.py]: Done

File: RUFAS/routines/manure/manure_treatments/composting.py
Comment (allisterakun) [reply RUFAS/routines/manure/manure_treatments/composting.py]: Done

File: RUFAS/routines/manure/manure_treatments/composting.py
Comment (allisterakun) [reply RUFAS/routines/manure/manure_treatments/composting.py]: Done

**Review by ltp23 — APPROVED** [review]
LGTM! Thanks for doing the updates, Allister! 

One other thing to consider is to add some error handling in the calculations in case of negative numbers (known to happen at other places in the manure module) or something  like that. SMEs may give better feedback on this.

**Comment by KFosterReed** [comment]
Hi Allister! This is looking really good! I made some changes to the nitrogen naming and reporting (and snuck in a new output in the Manure Handler Outputs that is helping me with the review :) ). I am also not sure if the changes I made to the organic/inorganic nitrogen function inputs go against recommended coding practices but it made it easier for me to read and seems to make sense so that we don't have to call the function to calculate the total N each time. For the time being, this allows me to review the N emission outputs

**Comment by allisterakun** [comment]
> Hi Allister! This is looking really good! I made some changes to the nitrogen naming and reporting (and snuck in a new output in the Manure Handler Outputs that is helping me with the review :) ). I am also not sure if the changes I made to the organic/inorganic nitrogen function inputs go against recommended coding practices but it made it easier for me to read and seems to make sense so that we don't have to call the function to calculate the total N each time. For the time being, this allows me to review the N emission outputs

These changes make sense! Thank you Kristan!

I'll update the tests and listing issues in a bit!

**Comment by KFosterReed** [comment]
Hi Allister! I made a couple of small changes (in the code and design doc) but after the tests updated and metadata/input files are in are in alignment with the main branch it looks good to me!! I'm going to approve it so once those small changes are fixed you can merge :)

**Review by KFosterReed — APPROVED** [review]

**Pull request merged**
